docs/AVS-Guide.md

```diff
@@ -0,0 +1,85 @@
+
+## Integration with EigenLayer Contracts:
+In this section, we will explain various API interfaces that EigenLayer provides which are essential for AVSs to integrate with EigenLayer. 
+
+### *Opting into AVS*
+In order for any EigenLayer operator to be able to opt-in to a AVS, EigenLayer provides two interfaces: `optIntoSlashing(..)` and `recordFirstStakeUpdate(..)`. The sequential flow for opting into a AVS using these functions is as follows:
+1. The operator first opts into slashing by calling  `optIntoSlashing(..)` in `Slasher.sol`, where it has to specify the address of the AVS's ServiceManager contract in the argument. This step results in the operator giving permission to the AVS's ServiceManager contract to slash the operator via EigenLayer, if the operator is ever proven to have engaged in adversarial behavior while responding to the AVS's task. Successful call to  `optIntoSlashing(..)` emits the `OptedIntoSlashing(..)` event.
+2. Next, the operator needs to register with the AVS on chain via a AVS-specific registry contract (see [this](https://github.com/Layr-Labs/eigenlayer-contracts/blob/middleware-guide/docs/Middleware-Guide.md#guide-to-provided-middleware-contracts) section for examples). To integrate with EigenLayer, the AVS's Registry contract provides a registration endpoint that calls on the `recordFirstStakeUpdate(..)` in the AVS's ServiceManager contract which in turn calls the `recordFirstStakeUpdate(..)` function in `Slasher.sol`. On successful execution of this function call, the event `MiddlewareTimesAdded(..)` is emitted and the operator has to start serving the tasks from the AVS.
+
+The following figure illustrates the above flow: 
+![Operator opting-in](./images/operator_opting.png)
+
+### *Recording Stake Updates*
+EigenLayer is a dynamic system where stakers and operators are constantly adjusting amounts of stake delegated via the system. It is therefore imperative for an AVS to be aware of any changes to stake delegated to its operators. In order to facilitate this, EigenLayer offers the `recordStakeUpdate(..)` API in the EigenLayer Slasher contract.
+
+Let us illustrate the usage of this facility with an example: A staker who has opted-in to an AVS via delegation, wants to partially withdraw some of its stake. Whenever the staker partially withdraw some of its stake, it affects all the AVSs uniformly that its operator is participating in. The series of steps for partially withdrawing stake is as follows:
```

Comment:
I find this wording unclear. Suggest changing:
FROM
"A staker who has opted-in to an AVS via delegation, wants to partially withdraw some of its stake. Whenever the staker partially withdraw some of its stake, it affects all the AVSs uniformly that its operator is participating in. The series of steps for partially withdrawing stake is as follows:"

TO
"A staker has delegated to an operator, who has opted-in to serving an AVS. Whenever the staker withdraws some or all of its stake from EigenLayer, this withdrawal affects all the AVSs uniformly that the staker's delegated operator is participating in. The series of steps for withdrawing stake is as follows:"